### PR TITLE
Add tests for r or s zero in p256 ecdsa

### PR DESCRIPTION
**Pull request by jonas-lj**
In the two secp256r1 ECDSA schemes it is not possible to deserialize a signature where either s or r are zero, so the implementation is working but it was a bit unclear in the code how this is handled. This PR adds tests and a few comments to clarify.

### CODE REVIEW

File: fastcrypto/src/secp256r1/recoverable.rs [hunk 1]
```diff
@@ -206,30 +212,31 @@ impl RecoverableSignature for Secp256r1RecoverableSignature {
...
-            );
-
-            Ok(Secp256r1PublicKey {
-                pubkey: VerifyingKey::from_affine(pk)
-                    .map_err(|_| FastCryptoError::GeneralOpaqueError)?,
-                bytes: OnceCell::new(),
-            })
-        } else {
-            Err(FastCryptoError::GeneralOpaqueError)
+        if big_r.is_none().into() {
```
Comment (jonas-lj) [inline fastcrypto/src/secp256r1/recoverable.rs:215]: Move the return closer to the check to make the code more readable.

**Review by benr-ml — APPROVED** [review]
thanks!

**Review by kchalkias — APPROVED** [review]
